src/v/iceberg/manifest_entry_type.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_entry_type.h"
+
+namespace iceberg {
+
+namespace {
+
+struct_type data_file_type(partition_key_type partition_type) {
+    struct_type r2_type;
+    // TODO: somehow make this not a magic number.
+    r2_type.fields.reserve(16);
+    r2_type.fields.emplace_back(
+      nested_field::create(134, "content", field_required::yes, int_type()));
+    r2_type.fields.emplace_back(nested_field::create(
+      100, "file_path", field_required::yes, string_type()));
+    r2_type.fields.emplace_back(nested_field::create(
+      101, "file_format", field_required::yes, string_type()));
+    r2_type.fields.emplace_back(nested_field::create(
+      102, "partition", field_required::yes, std::move(partition_type.type)));
+    r2_type.fields.emplace_back(nested_field::create(
+      103, "record_count", field_required::yes, long_type()));
+    r2_type.fields.emplace_back(nested_field::create(
+      104, "file_size_in_bytes", field_required::yes, long_type()));
+    r2_type.fields.emplace_back(nested_field::create(
+      108,
+      "column_sizes",
+      field_required::no,
+      map_type::create(
+        117, int_type(), 118, field_required::yes, long_type())));
+    r2_type.fields.emplace_back(nested_field::create(
+      109,
+      "value_counts",
+      field_required::no,
+      map_type::create(
+        119, int_type(), 120, field_required::yes, long_type())));
```

Comment:
wow what are these numbers? indices or something?

Reply:
These are field ids (basically Iceberg's column ids) https://iceberg.apache.org/spec/#manifests and https://github.com/redpanda-data/redpanda/blob/dev/src/v/iceberg/avroschemas/manifest_entry.schema.json